cmd/kubeadm/app/preflight/checks.go

```diff
@@ -1002,6 +1003,17 @@ func RunJoinNodeChecks(execer utilsexec.Interface, cfg *kubeadmapi.NodeConfigura
 			criCtlChecker)
 	}
 
+	for _, server := range cfg.DiscoveryTokenAPIServers {
+		ipstr, portstr, err := net.SplitHostPort(server)
+		if err == nil {
+			if port, err := strconv.Atoi(portstr); err == nil {
```

Comment:
I am trying to understand, why can't this `if` check be part of the existing loop below (originally on line 1005)?

Reply:
I did it for better readability. Existing loop adds FileContentCheck only  once. It's possible to add this code there, but we'll have to skip adding FileContentCheck if it's already added. This would make the code in the loop less readable in my opinion.

Reply:
I could only come up with this, which I consider less readable. Please, let me know if you prefer this or have better suggestion.

```
        fccheckAdded := false
        for _, server := range cfg.DiscoveryTokenAPIServers {
                ipstr, portstr, err := net.SplitHostPort(server)
                if err == nil {
                        if port, err := strconv.Atoi(portstr); err == nil {
                                checks = append(checks,
                                        HTTPProxyCheck{Proto: "https", Host: ipstr, Port: port},
                                )
                        }
                        if !fccheckAdded {
                                if ip := net.ParseIP(ipstr); ip != nil {
                                        if ip.To4() == nil && ip.To16() != nil {
                                                checks = append(checks,
                                                        FileContentCheck{Path: bridgenf6, Content: []byte{'1'}},
                                                )
                                                fccheckAdded = true // Ensure that check is added only once
                                        }
                                }
                        }
                }
        }
```